_javascripts/page-loader.js

```diff
@@ -1,114 +1,68 @@
+  "openshift-gitops": "https://docs.openshift.com/gitops/",
+  "openshift-origin": "https://docs.okd.io/",
+  "openshift-pipelines": "https://docs.openshift.com/pipelines/",
+  "openshift-serverless": "https://docs.openshift.com/serverless/",
+};
+
+// Helper function to confirm asynchronously
+async function confirmAsync(message) {
+  return window.confirm(message);
 }
 
 function versionSelector(list) {
+  "use strict";
 
-  // the version we want
   newVersion = list[list.selectedIndex].value;
+  currentVersion = window.location.pathname.split("/")[2];
 
-  // spilt the current path
-  var pathArray = window.location.pathname.split( '/' );
-
-  // so we can get the current version
-  currentVersion = pathArray[2];
+  let baseUrl = urlMappings[dk];
 
-  // if switching major versions, just take the user to the main landing page
-  // as files change a lot between major versions.
+  //Handle special OCP case
+  if (["3.0", "3.1", "3.2"].includes(newVersion) && dk === "openshift-enterprise") {
+    baseUrl = "https://docs.openshift.com/enterprise/";
+  }
 
-  if(currentVersion.charAt(0) === newVersion.charAt(0)) {
-    // the file path is just the version number + the end of the path
-    fileRequested =
-      window.location.pathname.substring(
-        window.location.pathname.lastIndexOf(currentVersion) +
-        currentVersion.length);
+  if (currentVersion.charAt(0) === newVersion.charAt(0)) {
+    const versionIndex = window.location.pathname.lastIndexOf(currentVersion) + currentVersion.length;
+    fileRequested = window.location.pathname.substring(versionIndex);
   } else {
     fileRequested = "/welcome/index.html";
   }
 
-  // alert(fileRequested);
-
-  // in 3.3 and above, we changed to container-platform
-  if(['3.0', '3.1', '3.2'].contains(newVersion)) {
-    newLink = "https://docs.openshift.com/enterprise/" +
-      newVersion +
-      fileRequested;
-  } else if (['3.65', '3.66', '3.67', '3.68', '3.69', '3.70', '3.71', '3.72', '3.73', '3.74', '4.0', '4.1'].contains(newVersion)) {
-    // check and handle links for RHACS versions
-    newLink = "https://docs.openshift.com/acs/" +
-      newVersion +
-      fileRequested;
-  } else if (['1.28', '1.29'].contains(newVersion)) {
-    // check and handle links for Serverless versions
-    newLink = "https://docs.openshift.com/serverless/" +
-      newVersion +
-      fileRequested;
-  } else if (['1.8', '1.9'].contains(newVersion)) {
-    // check and handle links for GitOps versions
-    newLink = "https://docs.openshift.com/gitops/" +
-      newVersion +
-      fileRequested;
-  } else {
-    //if distro key is openshift enterprise
-    if (dk == "openshift-enterprise") {
-    newLink = "https://docs.openshift.com/container-platform/" +
-      newVersion +
-      fileRequested;
-    }
-    else if (dk == "openshift-origin"){
-      newLink = "https://docs.okd.io/" +
-      newVersion +
-      fileRequested;
-    }
-  }
+  newLink = `${baseUrl}${newVersion}${fileRequested}`;
```

Comment:
You need to handle the enterprise 3.0,3.1,3.2 special case here too, right?

Reply:
WE arleady set the baseUrl value before this instance, so it is already covered.